Tasks/AzureResourceGroupDeploymentV2/operations/DeploymentGroupExtensionHelper.ts

```diff
@@ -30,9 +29,15 @@ export class DeploymentGroupExtensionHelper {
         var listOfVms: az.VM[] = await this.azureUtils.getVMDetails();
         var extensionAddedOnVMsPromises: Promise<any>[] = [];
         for (var vm of listOfVms) {
-            extensionAddedOnVMsPromises.push(this.addExtensionOnSingleVM(vm));
+            extensionAddedOnVMsPromises.push(this.addExtensionOnSingleVM(vm).catch(err => err));
         }
-        await Promise.all(extensionAddedOnVMsPromises);
+
+        var extensionErrorForVMs: any[] = await Promise.all(extensionAddedOnVMsPromises);
+        var errorString: string = utils.buildErrorString(extensionErrorForVMs);
+        if (!!errorString) {
+            throw new Error(tl.loc("DeploymentGroupConfigurationNotSucceeded", errorString));
```

Comment:
Does this expand all the elements of the array

Reply:
I have added buildErrorString() function to take non-null strings from array and concatenate them.